src/java/org/apache/cassandra/service/accord/AccordSafeCommandStore.java

```diff
@@ -341,6 +343,7 @@ public RangesForEpoch ranges()
 
     private <O> O mapReduce(Routables<?> keysOrRanges, BiFunction<CommandsSummary, O, O> map, O accumulate)
     {
+        Invariants.checkState(context.keys().containsAll(keysOrRanges));
```

Comment:
```suggestion
        Invariants.checkState(context.keys().containsAll(keysOrRanges), "Attempted to access keysOrRanges outside of what was asked for; asked for %s, accessed %s", context.keys(), keysOrRanges);
```

previous logic would tell you what ranges you were missing, we shouldn't loose that detail